### ISSUE TITLE
0.4.12 compiler throws an "Internal compiler error" when it should throw "Undeclared identifier"

### ISSUE THREAD

**Issue by Dexaran** [ethereum/solidity#2533]
I'm trying to compile this code:

```js
pragma solidity ^0.4.11;


contract TEST_1
{
    //address test;
    function foo()
    {
        test.delegatecall(bytes4(sha3("call_TEST_3()")));
    }
}
```

0.4.12 compiler throws this error:

```
Internal exception in StandardCompiler::compileInternal: /src/libsolidity/ast/ASTJsonConverter.cpp(791): Throw in function string dev::solidity::ASTJsonConverter::functionCallKind(dev::solidity::FunctionCallKind)
Dynamic exception type: N5boost16exception_detail10clone_implIN3dev8solidity21InternalCompilerErrorEEE
std::exception::what: std::exception
[PN3dev11tag_commentE] = Unknown kind of function call .
```

It's obviously that compiler should throw "Undeclared identifier" when I'm trying to call an undeclared identifier.

0.4.11 compiler throws a correct error type.

**Comment by chriseth** [comment]
This has been fixed in 0.4.13.

**Comment by Dexaran** [comment]
@chriseth `0.4.13-nightly.2017.7.5+commit.2b505e7a` throws internal compiler error too.

EDIT: I see `0.4.13-nightly.2017.7.6+commit.40d4ee49` is showing correct error now.
